src/arithmetic/arithmetic_expression_construct.c

```diff
@@ -116,8 +116,14 @@ static AR_ExpNode *_AR_EXP_FromIdentifierExpression(const cypher_astnode_t *expr
 }
 
 static AR_ExpNode *_AR_EXP_FromIdentifier(const cypher_astnode_t *expr) {
-	// check if the identifier is a named path identifier
 	AST *ast = QueryCtx_GetAST();
+	if(ast == NULL) {
+		// Attempted to access the AST before it has been constructed.
+		ErrorCtx_SetError("Attempted to access variable before it has been defined");
+		return AR_EXP_NewConstOperandNode(SI_NullVal());
+	}
```

Comment:
Please extend this comment, explaining how is it possible to reach this situation,
give an example query, e.g. `CYPHER param=[a] MATCH (a) RETURN param`.